File: app/src/main/java/com/aivoice/translate/adpaters/LanguageAdapter.java
package com.aivoice.translate.adpaters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView.Adapter;
import androidx.recyclerview.widget.RecyclerView.ViewHolder;

import com.aivoice.translate.Activity.LanguageActivity;
import com.aivoice.translate.Activity.TextActivity;
import com.aivoice.translate.Activity.VoiceActivity;
import com.aivoice.translate.R;
import com.aivoice.translate.model.LanguagesModel;

import java.util.ArrayList;
import java.util.List;

import static com.aivoice.translate.utils.Utils.WHATIS;

public class LanguageAdapter extends Adapter<LanguageAdapter.ExampleViewHolder> {
    private ArrayList<LanguagesModel> exampleList = new ArrayList<LanguagesModel>();

    class ExampleViewHolder extends ViewHolder {
        TextView mTxtLang;

        ExampleViewHolder(View itemView) {
            super(itemView);
            this.mTxtLang = itemView.findViewById(R.id.mTxtLang);

        }
    }

    public LanguageAdapter(List<LanguagesModel> exampleList2) {
        exampleList.clear();
        this.exampleList.addAll(exampleList2);
    }

    public ExampleViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ExampleViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_language, parent, false));
    }

    public void onBindViewHolder(ExampleViewHolder holder, int position) {
        final LanguagesModel currentItem = this.exampleList.get(position);
        holder.mTxtLang.setText(currentItem.getLanguageName());
        holder.mTxtLang.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if (WHATIS.equalsIgnoreCase("text")) {
                    if (TextActivity.getInstance() != null)
                        TextActivity.getInstance().mGetSelctedLang(currentItem.getLanguageName(), currentItem.getLanguageCode());
                } else {
                    if (VoiceActivity.getInstance() != null)
                        VoiceActivity.getInstance().mGetSelctedLang(currentItem.getLanguageName(), currentItem.getLanguageCode());
                }
                LanguageActivity.getInstance().finish();

            }
        });
    }

    public int getItemCount() {
        return this.exampleList.size();
    }

    public void setFilter(List<LanguagesModel> filterdNames) {
        exampleList.clear();
        this.exampleList.addAll(filterdNames);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/texttranslate/voiceimage/adpaters/LanguageAdapter.kt
package com.texttranslate.voiceimage.adpaters

import android.annotation.SuppressLint
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.texttranslate.voiceimage.R
import com.texttranslate.voiceimage.adpaters.LanguageAdapter.LanguageViewHolder
import com.texttranslate.voiceimage.interfaces.OnItemClickListener
import com.texttranslate.voiceimage.model.LanguagesModel

class LanguageAdapter(exampleList2: List<LanguagesModel>?, listener: OnItemClickListener?) :
    RecyclerView.Adapter<LanguageViewHolder>() {
    private val exampleList = ArrayList<LanguagesModel>()
    var mListener: OnItemClickListener?

    init {
        exampleList.clear()
        exampleList.addAll(exampleList2!!)
        mListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LanguageViewHolder {
        return LanguageViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_language, parent, false)
        )
    }

    override fun onBindViewHolder(holder: LanguageViewHolder, position: Int) {
        val currentItem = exampleList[position]
        holder.mTxtLang.text = currentItem.languageName

        holder.mTxtLang.setOnClickListener {
            Log.d("TAGAdapter", "onclick: ")
            if (mListener != null) {
                mListener!!.mGetSelctedLang(currentItem.languageName, currentItem.languageCode)
            }
        }
    }

    override fun getItemCount(): Int {
        return exampleList.size
    }

    class LanguageViewHolder internal constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var mTxtLang: TextView

        init {
            mTxtLang = itemView.findViewById(R.id.mTxtLang)
        }
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setFilter(filterdNames: List<LanguagesModel>?) {
        exampleList.clear()
        exampleList.addAll(filterdNames!!)
        notifyDataSetChanged()
    }
}